### PR TITLE
Test solve

### PR DESCRIPTION
**Pull request by ZHG2017**
Having made a copy for the solve function to avoid changing the original coefficient matrix.

### CODE REVIEW

File: tests/test-solve.C [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	
+	return ok;
+}
+
+int main(int argc, char** argv)
+{
+	cerr<<setprecision(10);
+	Givaro::Integer q=-1;
+	size_t b=0;
+	size_t m=6;
```
Comment (ClementPernet) [inline tests/test-solve.C]: Use a larger default dimension, like 300 or so.

File: tests/test-solve.C [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+	size_t iters=100;
```
Comment (ClementPernet) [inline tests/test-solve.C]: and reduce the number of iterations, like 3

File: tests/test-solve.C [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                END_OF_ARGUMENTS
+        };
+
+	FFLAS::parseArguments(argc,argv,as);
+
+	bool ok = true;
+
+	do{
+		ok = ok && run_with_field<Modular<double> >(q,b,m,iters,seed);
+
```
Comment (ClementPernet) [inline tests/test-solve.C]: remove empty lines

File: tests/test-solve.C [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+		ok = ok && run_with_field<ModularBalanced<double> >(q,b,m,iters,seed);
+
+		ok = ok && run_with_field<Modular<float> >(q,b,m,iters,seed);
+
+		ok = ok && run_with_field<ModularBalanced<float> >(q,b,m,iters,seed);
+		ok = ok && run_with_field<Modular<int32_t> >(q,b,m,iters,seed);
+		ok = ok && run_with_field<ModularBalanced<int32_t> >(q,b,m,iters,seed);
+		ok = ok && run_with_field<Modular<int64_t> >(q,b,m,iters,seed);
+		ok = ok && run_with_field<ModularBalanced<int64_t> >(q,b,m,iters,seed);
+
```
Comment (ClementPernet) [inline tests/test-solve.C]: missing specializations for Modular<Givaro::Integer> with fixed bitsize ( 5) and with bit size set to b. See test-lu.C for example.

File: tutorials/ffpack-solve.C [hunk 5]
```diff
@@ -1,4 +1,8 @@
+<<<<<<< HEAD
```
Comment (ClementPernet) [inline tutorials/ffpack-solve.C]: remove merge failure signs

File: tutorials/ffpack-solve.C [hunk 6]
```diff
@@ -1,4 +1,8 @@
...
+/* Copyright (c) FFLAS-FFPACK 
+=======
```
Comment (ClementPernet) [inline tutorials/ffpack-solve.C]: also here

File: tutorials/ffpack-solve.C [hunk 7]
```diff
@@ -1,4 +1,8 @@
...
 /* Copyright (c) FFLAS-FFPACK
+>>>>>>> master
```
Comment (ClementPernet) [inline tutorials/ffpack-solve.C]: and here

File: tutorials/Makefile.am [hunk 8]
```diff
@@ -35,21 +35,19 @@ TUTORIALS = fflas_101 fflas_101_lvl1 fflas-101_1 fflas-101_3 2x2-fgemm 101-fgemm
 2x2_pluq_SOURCES = 2x2-pluq.C
 fflas_101_SOURCES = fflas_101.C
 fflas_101_lvl1_SOURCES = fflas_101_lvl1.C
-#fflas_101_lvl2_SOURCES = fflas_101_lvl2.C
+fflas_101_lvl2_SOURCES = fflas_101_lvl2.C
```
Comment (ClementPernet) [inline tutorials/Makefile.am]: this line should not be uncommented

File: tests/test-solve.C [hunk 9]
```diff
@@ -48,78 +48,58 @@ using namespace FFPACK;
...
-
-}
-
 
 template<typename Field, class RandIter>
 bool check_solve(const Field &F, size_t m, RandIter& Rand){
 
-	typedef typename Field::Element Element;
-	Element * A, *A2,  *B, *B2, *x;
+typedef typename Field::Element_ptr Element;
```
Comment (ClementPernet) [inline tests/test-solve.C]: Don't do that: it is misleading as the typedef'ed Element is not an alias to Field::Element but Field::Element_ptr.

File: tests/test-solve.C [hunk 10]
```diff
@@ -48,78 +48,58 @@ using namespace FFPACK;
...
 
 	bool ok = true;
 	if (FFLAS::fequal (F, m, 1, B2, incb, B, incb)){
 
 		cout << "PASSED ("<<time<<")"<<endl;
 
 	} else{
+		#ifdef DEBUG
 		FFLAS::WriteMatrix(std::cout<<"A*x:="<<std::endl,F,m,1,B2,incb)<<std::endl;
 		FFLAS::WriteMatrix(std::cout<<"b:="<<std::endl,F,m,1,B,incb)<<std::endl;
```
Comment (ClementPernet) [inline tests/test-solve.C]: Output should be `x = ....` and not `b:=`

**Comment by ZHG2017** [comment]
In fact here, I just print out the result after the multiplication between
matrix A and the output from function slove named as x, then I print out b
to see if A*x == b, not to check out the output x.

**Comment by ClementPernet** [comment]
Good to go.

**Pull request merged**